lib/_stream_writable.js

```diff
@@ -831,12 +775,10 @@ const destroy = destroyImpl.destroy;
 Writable.prototype.destroy = function(err, cb) {
   const state = this._writableState;
   if (!state.destroyed) {
-    for (let entry = state.bufferedRequest; entry; entry = entry.next) {
-      process.nextTick(entry.callback, new ERR_STREAM_DESTROYED('write'));
+    for (const { callback } of state.getBuffer()) {
```

Comment:
Perhaps this should explicitly use `this.buffered.slice(this.bufferedIndex)` as `getBuffer` may have been tampered with in userspace?